format/spec.md

```diff
@@ -702,6 +703,41 @@ Blob metadata is a struct with the following fields:
 | _optional_ | _optional_ | **`properties`** | `map<string, string>` | Additional properties associated with the statistic. Subset of Blob properties in the Puffin file. |
 
 
+#### Partition statistics
+
+Partition statistics files are based on [Partition Statistics file spec](#partition-statistics-file). Partition statistics are informational. A reader can choose to
+ignore partition statistics information. Partition statistics support is not required to read the table correctly.
+Each table snapshot may be associated with at most one partition statistic file and the table can contain many partition statistics files associated with different table snapshots.
+A writer can optionally write the partition statistics file during each write operation. If the statistics file is written for the specific snapshot,
+it must be accurate and must be registered in the table metadata file to be considered as a valid statistics file for the reader.
+
+Partition statistics files metadata within `partition-statistics` table metadata field is a struct with the following fields:
+
+| v1 | v2 | Field name | Type | Description |
+|----|----|------------|------|-------------|
+| _required_ | _required_ | **`snapshot-id`** | `long` | ID of the Iceberg table's snapshot the partition statistics file is associated with. |
+| _required_ | _required_ | **`statistics-file-path`** | `string` | Path of the partition statistics file. See [Partition Statistics file](#partition-statistics-file). |
+| _required_ | _required_ | **`max-data-sequence-number`** | `long` | Maximum data sequence number of the Iceberg table's snapshot the partition statistics was computed from. |
+
+#### Partition Statistics file
+
+Statistics information for every partition tuple is stored as a row in the **table default format**.
+These rows are sorted (in ascending manner with NULL FIRST) based on all partition columns from `partition` in the same order
```

Comment:
The partition spec here probably needs to be specified? All statistics are stored based on the current partition spec? or are we storing based on the unified spec as of that current snapshot. 

For example say i have a table with 2 Specs
1. Identity A
2. Identity B

And the current spec is "B"

Are we storing tuples
(A, B)

or are we storing only
(B) 

In the statistics file?

Reply:
(A,B)

It is full unified spec computed from `Partitioning#partitionType`. We coerce the partition from previous partition stats file while writing the stats for the current snapshot to support smooth partition evolution.

https://github.com/apache/iceberg/blob/2b189759a902cdd8ad433d0c1643376b78ec867c/core/src/main/java/org/apache/iceberg/Partitioning.java#L242

Reply:
This needs to be in the spec then

Reply:
added a detailed note.